gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala

```diff
@@ -292,6 +310,7 @@ object ElCompiler extends StrictLogging {
   private val DateFormatRegex = """[^#{}()]+""".r
   private val NumberRegex = """\d+""".r
   private val NumberRegexWithNegative = """-?\d+""".r
+  private val FloatingRegexWithNegative = """[+-]?(\d*\.)?\d+""".r
```

Comment:
This regex is not correct:
* it matches incorrect values, eg `.1` or `0.1.1`
* why do you allow for `+` while NumberRegexWithNegative doesn't 

What about `-?\d+(\.\d+)?`?

Reply:
I tried a couple before I settled to this. Still made a boo boo when I pushed. Here is the list

1. `+.14` and `.14` are both valid  strings representing floating points, using `-?\d+(\.\d+)?` fails both on the `+` and the missing `0` before the decimal
2. Earlier I had the start and end anchors on the the regex, but that would fail with the parser, I've tried the start anchor `^` and this seems to work with the regex I've used - `^[+-]?(\d*\.)?\d+` (matches `1.1` from `1.1.1`)

As I see it `+145` should not fail for integers as this is a valid string representing an integer/long. While the `.1` is a valid floating point and `".1".toDouble` gives a valid `0.1: scala.Double`

I definitely need to add more test cases here :)

Reply:
I see my confusion here :) with the `+` I'll try something soon